src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java

```diff
@@ -68,22 +72,28 @@
         return null;
     }
 
-    public void doBuild(StaplerRequest req, StaplerResponse rsp, @QueryParameter String job) throws IOException, ServletException {
+    public void doBuild(StaplerRequest req, StaplerResponse rsp, @QueryParameter String job, @QueryParameter TimeDuration delay) throws IOException, ServletException {
         LOGGER.log(Level.FINE, "build on {0}", job);
-        AbstractProject<?,?> p = project(job, req, rsp);
-        ParametersDefinitionProperty pp = p.getProperty(ParametersDefinitionProperty.class);
+        ParameterizedJobMixIn.ParameterizedJob p = project(job, req, rsp);
+        if (delay == null) {
+            delay = new TimeDuration(p.getQuietPeriod());
+        }
+        ParametersDefinitionProperty pp = ((Job<?,?>) p).getProperty(ParametersDefinitionProperty.class);
         if (pp != null) {
             LOGGER.fine("wrong kind");
             throw HttpResponses.error(HttpServletResponse.SC_BAD_REQUEST, "use buildWithParameters for this build");
         }
-        Jenkins.getInstance().getQueue().schedule(p, p.getDelay(req), getBuildCause(req));
+        Jenkins.getInstance().getQueue().schedule(p, delay.getTime(), getBuildCause(req));
         ok(rsp);
     }
 
-    public void doBuildWithParameters(StaplerRequest req, StaplerResponse rsp, @QueryParameter String job) throws IOException, ServletException {
+    public void doBuildWithParameters(StaplerRequest req, StaplerResponse rsp, @QueryParameter String job, @QueryParameter TimeDuration delay) throws IOException, ServletException {
         LOGGER.log(Level.FINE, "buildWithParameters on {0}", job);
-        AbstractProject<?,?> p = project(job, req, rsp);
-        ParametersDefinitionProperty pp = p.getProperty(ParametersDefinitionProperty.class);
+        ParameterizedJobMixIn.ParameterizedJob p = project(job, req, rsp);
+        if (delay == null) {
+            delay = new TimeDuration(p.getQuietPeriod());
+        }
+        ParametersDefinitionProperty pp = ((Job<?,?>) p).getProperty(ParametersDefinitionProperty.class);
         if (pp == null) {
             LOGGER.fine("wrong kind");
             throw HttpResponses.error(HttpServletResponse.SC_BAD_REQUEST, "use build for this build");
```

Comment:
This message (and the corresponding one in `build(…)` could be improved to say something like `Use the /build URL to build this job`.